app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.java

```diff
@@ -235,4 +259,140 @@ protected void openReportPage(String accountId, String accountUsername, String s
         intent.putExtra("status_content", HtmlUtils.toHtml(statusContent));
         startActivity(intent);
     }
+
+    // https://mastodon.foo.bar/@User
+    // https://mastodon.foo.bar/@User/43456787654678
+    // https://pleroma.foo.bar/users/User
+    // https://pleroma.foo.bar/users/43456787654678
+    // https://pleroma.foo.bar/notice/43456787654678
+    // https://pleroma.foo.bar/objects/d4643c42-3ae0-4b73-b8b0-c725f5819207
+    private static boolean looksLikeMastodonUrl(String urlString) {
+        URI uri;
+        try {
+            uri = new URI(urlString);
+        } catch (URISyntaxException e) {
+            return false;
+        }
+
+        if (uri.getQuery() != null ||
+                uri.getFragment() != null ||
+                uri.getPath() == null) {
+            return false;
+        }
+
+        String path = uri.getPath();
+        return path.matches("^/@[^/]*$") ||
+                path.matches("^/users/[^/]+$") ||
+                path.matches("^/(@|notice)[^/]*/\\d+$") ||
+                path.matches("^/objects/[-a-f0-9]+$");
+    }
+
+    private void onBeginSearch(@NonNull String url) {
+        searchUrl = url;
+        showQuerySheet();
+    }
+
+    private boolean getCancelSearchRequested(@NonNull String url) {
+        return !url.equals(searchUrl);
+    }
+
+    private boolean getIsSearching() {
```

Comment:
I'm kinda late for this but why `get` prefix? Kotlin is the only reason I can think of but it's private anyway

Reply:
Good point, this naming is just left over from earlier iterations.